picard/album.py

```diff
@@ -163,7 +163,21 @@ def _release_request_finished(self, document, http, error):
         self.load_task = None
         parsed = False
         try:
-            if error:
+            if (error
+                and config.setting['track_ars']
+                and not self.gateway_timeout
+                and repr(http.attribute(QtNetwork.QNetworkRequest.HttpStatusCodeAttribute)) == "504"):
+                # Gateway timeout error on large release with track relationships
+                # reissue as a priority request without the track relationships.
+                self.gateway_timeout = True
+                log.debug("Server timeout for release %r: retrying request without track relationships", self.id)
```

Comment:
This should be a log.warning().

Imho it should be a user dialog, explaining it failed due to timeout and huge amount of data, and let the user chooses between: Retry, Retry without track relationships, Cancel
This can be done later.